File: ThirtyDaysChallange/Day1/SingleNumber.java
class Solution {
    public int singleNumber(int[] nums) {
        // Hashe set
        Set<Integer> seen = new HashSet<Integer>(nums.length);
        for(int i: nums){
            if(!seen.add(i)){
                seen.remove(i);           
            }           
        }
        return seen.iterator().next();
    }

    public int singleNumber2(int[] nums) {
        // Bit XOR
        int ans = 0;
        for(int i: nums){
            ans ^= i;
        }
        return ans;
    }
}

  

File: ThirtyDaysChallange/Day1/Single_Number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # Set
        seen = set()
        for i in nums:
            if i not in seen:
                seen.add(i)
            else:
                seen.remove(i)
        return seen.pop()

    def singleNumber2(self, nums: List[int]) -> int:
        # Bit XOR
        ans = 0
        for num in nums:
            ans ^= num
        return ans
